### PR TITLE
tvg_loader: code refactoring.

### PR DESCRIPTION
**Pull request by hermet**
keep it neaat & clean code for readibility and maintenence.

### CODE REVIEW

File: src/loaders/tvg/tvgTvgLoadParser.cpp [hunk 1]
```diff
@@ -384,98 +402,80 @@ static LoaderResult _parseShapeProperty(tvgBlock block, Shape *shape)
...
 
-    Paint* paint;
-    auto result = _parsePaint(block, &paint);
-    if (result == LoaderResult::Success) {
-        if (picture->paint(unique_ptr<Paint>(paint)) != Result::Success) {
-            return LoaderResult::LogicalCorruption;
-        }
+    if (_paintProperty(block)) return _parsePaintProperty(block, picture);
+
+    if (picture->paint(unique_ptr<Paint>(_parsePaint(block))) != Result::Success) {
```
Comment (mmaciola) [inline src/loaders/tvg/tvgTvgLoadParser.cpp:438]:
```
auto chlild = _parsePaint(block);
if (chlild && picture->paint(unique_ptr<Paint>(chlild)) != Result::Success) {
    delete(child);
```

File: src/loaders/tvg/tvgTvgLoadParser.cpp [hunk 2]
```diff
@@ -384,98 +402,80 @@ static LoaderResult _parseShapeProperty(tvgBlock block, Shape *shape)
...
+        return LoaderResult::LogicalCorruption;
     }
-    return result;
+
+    return LoaderResult::Success;
```
Comment (mmaciola) [inline src/loaders/tvg/tvgTvgLoadParser.cpp:442]: `return LoaderResult::InvalidType;`

File: src/loaders/tvg/tvgTvgLoadParser.cpp [hunk 3]
```diff
@@ -384,98 +402,80 @@ static LoaderResult _parseShapeProperty(tvgBlock block, Shape *shape)
...
+        default: return nullptr;
     }
-    return LoaderResult::Success;
+
+    auto ptr = baseBlock.data;
+
+    while (ptr < baseBlock.end) {
+        auto block = _readBlock(ptr);
+        if (block.end > baseBlock.end) return paint;
+        if (parser(block, paint) != LoaderResult::Success) return paint;
```
Comment (mmaciola) [inline src/loaders/tvg/tvgTvgLoadParser.cpp:475]:
```
auto result = parser(block, paint);
if (result != LoaderResult::Success && result != LoaderResult::InvalidType) return paint;
```

**Review by mmaciola — CHANGES_REQUESTED** [review]

File: src/loaders/tvg/tvgTvgLoadParser.cpp [hunk 4]
```diff
@@ -143,14 +160,13 @@ static LoaderResult _parseSceneProperty(tvgBlock block, Scene *scene)
...
 
-    Paint* paint;
-    auto result = _parsePaint(block, &paint);
-    if (result == LoaderResult::Success) {
-        if (scene->push(unique_ptr<Paint>(paint)) != Result::Success) {
-            return LoaderResult::MemoryCorruption;
-        }
+    if (_paintProperty(block)) return _parsePaintProperty(block, scene);
+
+    if (scene->push(unique_ptr<Paint>(_parsePaint(block))) != Result::Success) {
```
[the remark this answers is not in the document]
Comment (hermet) [reply src/loaders/tvg/tvgTvgLoadParser.cpp]: ok so you say returning nullptr is expected if new type is added later version...
fixed

File: src/loaders/tvg/tvgTvgLoadParser.cpp (hunk 2 above)
Comment (hermet) [reply src/loaders/tvg/tvgTvgLoadParser.cpp]: fixed

File: src/loaders/tvg/tvgTvgLoadParser.cpp (hunk 1 above)
Comment (hermet) [reply src/loaders/tvg/tvgTvgLoadParser.cpp]: fixed

File: src/loaders/tvg/tvgTvgLoadParser.cpp (hunk 3 above)
Comment (hermet) [reply src/loaders/tvg/tvgTvgLoadParser.cpp]: fixed

File: src/loaders/tvg/tvgTvgLoadParser.cpp [hunk 5]
```diff
@@ -143,14 +160,13 @@ static LoaderResult _parseSceneProperty(tvgBlock block, Scene *scene)
...
+        return LoaderResult::MemoryCorruption;
     }
-    return result;
+
+    return LoaderResult::Success;
```
[the remark this answers is not in the document]
Comment (hermet) [reply src/loaders/tvg/tvgTvgLoadParser.cpp]: There is no point actually since returning InvalidType is also successive case...
but anyhow fixed it.

File: src/loaders/tvg/tvgTvgLoadParser.cpp [hunk 6]
```diff
@@ -114,9 +114,7 @@ static LoaderResult _parseCmpTarget(const char *ptr, const char *end, Paint *pai
     auto cmpBlock = _readBlock(ptr);
     if (cmpBlock.end > end) return LoaderResult::SizeCorruption;
 
-    if (paint->composite(unique_ptr<Paint>(_parsePaint(cmpBlock)), cmpMethod) != Result::Success) {
-        return LoaderResult::MemoryCorruption;
-    }
+    paint->composite(unique_ptr<Paint>(), cmpMethod);
```
Comment (mmaciola) [inline src/loaders/tvg/tvgTvgLoadParser.cpp:117]: Undo the change. There it was ok.

File: src/loaders/tvg/tvgTvgLoadParser.cpp [hunk 7]
```diff
@@ -162,11 +160,12 @@ static LoaderResult _parseScene(tvgBlock block, Paint *paint)
 
     if (_paintProperty(block)) return _parsePaintProperty(block, scene);
 
-    if (scene->push(unique_ptr<Paint>(_parsePaint(block))) != Result::Success) {
-        return LoaderResult::MemoryCorruption;
+    if (auto paint = _parsePaint(block)) {
+        scene->push(unique_ptr<Paint>(paint));
```
Comment (mmaciola) [inline src/loaders/tvg/tvgTvgLoadParser.cpp:164]: Not mandatory but I suggest add failure checking 
`if (scene->push(unique_ptr<Paint>(paint)) != Result::Success) return LoaderResult::MemoryCorruption;`

File: src/loaders/tvg/tvgTvgLoadParser.cpp [hunk 8]
```diff
@@ -435,11 +434,12 @@ static LoaderResult _parsePicture(tvgBlock block, Paint* paint)
 
     if (_paintProperty(block)) return _parsePaintProperty(block, picture);
 
-    if (picture->paint(unique_ptr<Paint>(_parsePaint(block))) != Result::Success) {
-        return LoaderResult::LogicalCorruption;
+    if (auto paint = _parsePaint(block)) {
+        picture->paint(unique_ptr<Paint>(paint));
```
Comment (mmaciola) [inline src/loaders/tvg/tvgTvgLoadParser.cpp:438]: Same as above

**Review by mmaciola — CHANGES_REQUESTED** [review]
Please have a look

File: src/loaders/tvg/tvgTvgLoadParser.cpp (hunk 6 above)
Comment (hermet) [reply src/loaders/tvg/tvgTvgLoadParser.cpp:117]: @mmaciola it's my intention. not so useful.

File: src/loaders/tvg/tvgTvgLoadParser.cpp (hunk 7 above)
Comment (hermet) [reply src/loaders/tvg/tvgTvgLoadParser.cpp:164]: it's my intention. not so useful.

File: src/loaders/tvg/tvgTvgLoadParser.cpp (hunk 6 above)
Comment (mmaciola) [reply src/loaders/tvg/tvgTvgLoadParser.cpp:117]: please fix it: target for composite is missing there.
The previous code also handled nullptr from _parsePaint that here signifies an error.
